CodeChallenge/Challenges/BulletMatch/BulletMatch.swift

```diff
@@ -0,0 +1,121 @@
+these markings to a gun thought to used to fire the
+original bullet.
+
+Given two markings, one from a spent cartridge and the other
+from a suspected weapon, both as Strings, return true if they match
+and false if they don't.
+
+EXAMPLES:
+Match, returns true
+"| |||  |   |" and
+"| |||  |   |"
+
+Match, returns 1 though one is rotated. Bullets can be rotated.
+Your solution should account for the possible rotation of bullets.
+"|| ||| | " and
+" | || |||"
+
+Doesn't match, returns false
+"| |||  |   |" and
+"||| |  |   |"
+
+Problem adapted from http://bit.ly/2h57Wxe
+*/
+
+struct BulletChallenge: CodeChallengeType {
+    typealias InputType = (bulletMarkings: String, gunMarkings: String)
+    typealias OutputType = Bool
+    
+    var title = "Bullet Challenge"
+    
+    var entries: [CodeChallengeEntry<BulletChallenge>] = [
+        bugKrushaBulletMatchEntry
+    ]
+    
+    func generateDataset() -> [(bulletMarkings: String, gunMarkings: String)] {
+        return generateMarkings(count: 15000, length: 15)
+    }
+    
+    func verifyOutput(_ output: Bool, forInput input: (bulletMarkings: String, gunMarkings: String)) -> Bool {
+        guard
+            let vd = verificationData[input.bulletMarkings]
+            else { return false }
+        return vd.1 == output
+    }
+}
+
+private var verificationData =  [
+    // double check these
+    "| |||  |   |": ("| |||  |   |", true),
+    "| |||  |  |": ("||| |  |   |", false),
+    "|| ||| | ": (" | || |||", true)
+]
+
+/**
+ Turns bullet markings around to mimic a rotated
+ bullet.
+ */
+private func rotate(marking: String, turns: Int) -> String {
+    
+    guard turns != 0 else { return marking }
+    let offset = marking.index(marking.startIndex, offsetBy: turns)
+    
+    let prefix = marking.substring(from: offset)
+    let suffix = marking.substring(to: offset)
+    
+    return prefix + suffix
+}
+
+
+fileprivate func generateMarkings(count: Int, length: Int) -> [(String, String)] {
+    let characters: [Character] = ["|", " "]
+    
+    var markings = [
+        ("| |||  |   |", "| |||  |   |"),
+        ("| |||  |  |", "||| |  |   |)"),
+        ("|| ||| | ", " | || |||")
+    ]
+    
+    for _ in 0..<count {
+        var bulletMarking = ""
+        
+        for _ in 0..<length {
+            bulletMarking.append(characters[rand(below: 2)])
+        }
+        
+        guard verificationData[bulletMarking] == nil else { continue }
+        var gunMarking = rotate(marking: bulletMarking,
+                              turns: rand(below: bulletMarking.characters.count))
+
+        
+        if rand(below: 2) == 0 {
+            gunMarking += "| "
+            
+            verificationData[bulletMarking] = (gunMarking, false)
+            markings.append((bulletMarking, gunMarking))
+            continue
+        }
+        
+        verificationData[bulletMarking] = (gunMarking, true)
+        markings.append((bulletMarking, gunMarking))
+    }
+    return markings
+}
```

Comment:
I don't like the dataset being random because then each test run is going to be run with different inputs, which feels like not the most empirical of comparisons. That may not matter for this particular problem, I haven't spent enough time with it to know, but I thought I'd bring it up.

The alternative would be to generate a dataset randomly as you've done here, and then write that out to a file and add that file to the repo and then just load the file here.